06-building-a-program-start-to-finish/solving-the-problem/output/app/class/invertedIndexClass.js

```diff
@@ -0,0 +1,202 @@
+    || typeof fileName !== 'string' ? this.indices : this.indices[fileName];
+  }
+
+  /**
+   * function to search through created indices
+   * @function
+  * @param {String} fileName uploaded valid JSON file
+  * @param {String} query word(s) or terms to search for
+  * @returns {array} returns array searchResults contained search indices
+  */
+  searchIndices(fileName, query) {
+    const searchResults = [];
+    const result = {};
+    let indices = {};
+
+    if (this.getIndices(fileName)) {
+      indices[fileName] = this.getIndices(fileName);
+    } else {
+      indices = this.indices;
+    }
+
+    Object.keys(indices).forEach((book) => {
+      query.split(' ').forEach((word) => {
+        if (Object.prototype.hasOwnProperty.call(indices[book], word)) {
+          if (!Object.prototype.hasOwnProperty.call(result, book)) {
+            result[book] = {
+              terms: {},
+              fileName: book
+            };
+          }
+          result[book].terms[word] = indices[book][word];
+        }
+      });
+
+      searchResults.push(result[book]);
+    });
+
+    if (typeof searchResults[0] === 'undefined') {
+      return false;
+    }
+
+    return searchResults;
+  }
+
+
+  /**
+   * function to validate user uploaded files
+   * @function
+   * @param {object} file
+   * @returns {boolean} return boolean for all test cases
+  */
+  validateFile(file) {
+    let isValid = file;
+
+    if (file.length > 0) {
+      for (let i = 0; i < file.length; i += 1) {
+        if (file[i] === undefined
+          || file[i].title === '' || file[i].text === ''
+          || !file[i].title || !file[i].text
+          || typeof file[i].text !== 'string') {
+          isValid = false;
+        }
+      }
+    } else isValid = false;
+
+    return isValid;
+  }
+
+  /**
+   * remove duplicates from word array
+   * @function
+   * @param {array} array
+   * @returns {array} array equivalent with unique items
+  */
+  getUnique(array) {
+    return array.filter((value, index, self) => {
+      return self.indexOf(value) === index;
+    });
+  }
+
+  /**
+   * format JSON object joining title and text values together
+   * @function
+   * @param {object} file
+   * @returns {array} formatted array
+  */
+  formatJSON(file) {
+    let text;
+    let title;
+    let newDoc = [];
+
+    for (let item in file) {
+      title = file.title;
+      text = file.text;
+      newDoc.push(`${title} ${text}`);
+    }
+
+    newDoc = newDoc.join(' ');
+    return newDoc;
+  }
+
+  /**
+   * trim, remove special characters, white spaces and special characters
+   * remove word duplicates
+   * @function
+   * @param {string} text document title and text
+   * @returns {Array} tokens
+  */
+  tokenize(text) {
+    let tokens = [];
+
+    tokens = this.formatJSON(text)
+      .trim()
+      .toLowerCase()
+      .match(/\w+/g)
+      .sort();
+      
+    tokens = this.getUnique(tokens);
+    return tokens;
+  }
+}
```

Comment:
There are a host of 'Eslint' violations on this file. Please ensure that you have set your 'Eslint' configuration properly. Please fix them.